packages/bpk-component-snippet/src/BpkSnippet.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { MouseEvent } from 'react';
+
+import { BpkButtonV2, BUTTON_TYPES } from '../../bpk-component-button';
+import BpkText, {
+  TEXT_STYLES,
+  type TextStyle,
+} from '../../bpk-component-text/src/BpkText';
+import { cssModules } from '../../bpk-react-utils';
+
+import type { ButtonType } from '../../bpk-component-button/src/BpkButtonV2/common-types';
+
+import STYLES from './BpkSnippet.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  src: string;
+  altText: string;
+  headline?: string | null;
+  subheading?: string | null;
+  bodyText?: string | null;
+  buttonText?: string | null;
+  bodyStyle?: typeof TEXT_STYLES.bodyDefault | typeof TEXT_STYLES.bodyLongform;
+  buttonStyle?: ButtonType;
+  headlineStyle?: TextStyle;
+  desktopLayout?: 'imageLeft' | 'imageRight' | 'vertical';
+  imageOrientation?: 'landscape' | 'portrait' | 'square';
+  onClick?: (event: MouseEvent<HTMLButtonElement & HTMLAnchorElement>) => void;
+};
+
+const BpkSnippet = ({
+  altText,
+  bodyStyle = TEXT_STYLES.bodyDefault,
+  bodyText,
+  buttonStyle = BUTTON_TYPES.primary,
+  buttonText,
+  desktopLayout = 'imageLeft',
+  headline,
+  headlineStyle = TEXT_STYLES.hero5,
+  imageOrientation,
+  onClick,
+  src,
+  subheading,
+}: Props) => (
+  <div
+    className={getClassName(
+      'bpk-snippet',
+      desktopLayout === 'imageRight' && 'bpk-snippet--row-reverse',
+      desktopLayout === 'vertical' && 'bpk-snippet--vertical',
+    )}
+  >
+    <div
+      className={getClassName(
+        desktopLayout === 'vertical'
+          ? 'bpk-snippet--vertical--container'
+          : 'bpk-snippet--container',
+        `bpk-snippet--image__${imageOrientation}`,
+      )}
+    >
+      <img
+        className={getClassName('bpk-snippet--image')}
+        alt={altText || ''}
+        src={src}
+        loading="lazy"
+      />
```

Comment:
Any reason not to use the `BpkImage` component here with our `withLazyLoading` HoC?

As this will take out a lot of manual work here for you with styles and setup

Reply:
This is referred to [bpk-content-cards](https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-content-cards/src/BpkContentCard.js#L56-L61) component, 
And the rounded corners of bpkimage do not support the size of the design.

Reply:
> And the rounded corners of bpkimage do not support the size of the design.

I am not quite sure what you mean here, if you use the Backpack Image component the rounded corners will adapt to the image providing you send the correct aspect ratio to the props of it, meaning you can then set the ratios either 

- Yourself and we can expose a property
- We can just do it via JS instead of CSS

Reply:
Replaced the native img with BpkImage